packages/server/src/modules/user/seed-users-with-books.command.ts

```diff
@@ -0,0 +1,518 @@
+            ),
+            userId: buyer.id,
+          })),
+      });
+
+      const reservations = await this.prisma.reservation.findMany({
+        where: {
+          userId: buyer.id,
+          request: {
+            userId: buyer.id,
+            saleId: null,
+          },
+          bookId: {
+            in: bookIds,
+          },
+        },
+      });
+
+      const cart = await this.prisma.cart.create({
+        data: {
+          items: {
+            create: reservations
+              .slice(0, reservations.length - 2)
+              .map(({ id, bookId }) => ({
+                fromReservationId: id,
+                bookId,
+              })),
+          },
+          userId: buyer.id,
+          createdById: buyer.id,
+          retailLocationId,
+        },
+        include: {
+          items: {
+            include: {
+              book: {
+                include: {
+                  copies: {
+                    where: {
+                      OR: [
+                        {
+                          sales: {
+                            none: {},
+                          },
+                        },
+                        {
+                          sales: {
+                            every: {
+                              refundedAt: {
+                                not: null,
+                              },
+                            },
+                          },
+                        },
+                      ],
+                    },
+                  },
+                },
+              },
+            },
+          },
+        },
+      });
+
+      const itemsToSell = cart.items.slice(0, cart.items.length - 2);
+
+      await this.prisma.sale.createMany({
+        data: itemsToSell.map((item) => ({
+          bookCopyId: item.book.copies[0].id,
+          cartCreatedById: buyer.id,
+          createdById: buyer.id,
+          purchasedAt: new Date(),
+          purchasedById: buyer.id,
+        })),
+      });
+
+      const sales = await this.prisma.sale.findMany({
+        where: {
+          bookCopy: {
+            book: {
+              retailLocationId,
+            },
+          },
+          purchasedById: buyer.id,
+          bookCopyId: {
+            in: itemsToSell.map(({ book }) => book.copies[0].id),
+          },
+        },
+        include: {
+          bookCopy: {
+            include: {
+              book: {
+                include: {
+                  requests: true,
+                  reservations: true,
+                },
+              },
+              sales: true,
+            },
+          },
+        },
+      });
+
+      // Connect requests and reservations to the sales
+      let i = 0;
+      for (const sale of sales) {
+        await this.#updateReservationsRequestsAndCart(sale, cart);
+
+        if (i % 3 === 2) {
+          await this.#refundBookCopy(sale, retailLocationId);
```

Comment:
Even if this is super fast at the moment, the correct way to handle this one is to create an array of promises.